### PR TITLE
Proposal for reducing number of ciphers

### PR DESCRIPTION
**Pull request by hannestschofenig**
I listened to the recording of the IETF #118 COSE working group meeting and based on the feedback I have cut down the number of ciphersuites.

### CODE REVIEW

**Review by selfissued — APPROVED** [review]

**Comment by dajiaji** [comment]
I feel that it would be better to keep the ciphersuites with ChaChaPoly (36,38 and 40) for keeping the symmetry of the choices, but I have no clear reason or use case to strongly argue for it. I agree with this PR.

File: draft-ietf-cose-hpke.md [hunk 1]
```diff
@@ -334,30 +334,30 @@ document and the values registered in the HPKE IANA registry {{HPKE-IANA}}.
...
+ciphersuites can be registered into the already existing registry.
+For example, once post-quantum cryptographic algorithms have been standardized
+it might be beneficial to register ciphersuites for use with COSE-HPKE.
+Additionally, ciphersuites utilizing the compact encoding of the public keys,
+as defined in {{I-D.irtf-cfrg-dnhpke}}, may be standardized for use in
+constrained environments.
+
+As a guideline for ciphersuite submissions to the IANA CoSE algorithm
+registry, the designated experts must only register combinations of 
+(KEM, KDF, AEAD) triple that consitute validate combinations for use with HPKE.
```
Comment (ilaril) [inline draft-ietf-cose-hpke.md:359]: "validate"?  Should this be "valid" or something else?

And that "must" might be better uppercased.

And other guidelines that might be considered:

- KDF used SHOULD (if possible) match one internally used by the KEM.
- Components SHOULD NOT be mixed between global and different national standards.

Comment (hannestschofenig) [inline draft-ietf-cose-hpke.md:359]:
```suggestion
(KEM, KDF, AEAD) triple that consitute valid combinations for use with HPKE, the KDF used should (if possible) match one internally used by the KEM, and components should not be mixed between global and national standards.
```

Comment (hannestschofenig) [reply draft-ietf-cose-hpke.md]: Thanks for the comments, Ilari. These additional guidelines are good.

Regarding the use of RFC 2119 language: I prefer to use the MAY, SHOULD, MUST terms when it concerns interoperability. Here we are just giving guidance to the experts and therefore I believe it is fine to use lower-case terms.

**Pull request merged**